tests/QE/test_xattrs.py

```diff
@@ -372,3 +374,434 @@ async def test_purge(self, cblpytest: CBLPyTest, dataset_path: Path) -> None:
+        sg_all_docs = await sg.get_all_documents(sg_db, "_default", "_default")
+        assert len(sg_all_docs.rows) == num_docs * 2, (
+            f"Expected {num_docs * 2} docs via SG, got {len(sg_all_docs.rows)}"
+        )
+
+        self.mark_test_step("Verify SG sees all docs via _changes")
+        changes = await sg.get_changes(sg_db, "_default", "_default")
+        assert len(changes.results) == num_docs * 2, (
+            f"Expected {num_docs * 2} changes via SG, got {len(changes.results)}"
+        )
+
+        self.mark_test_step(
+            f"Perform concurrent updates ({num_updates} per doc) from SDK and SG"
+        )
+
+        async def update_from_sg() -> None:
+            """Update documents from Sync Gateway side with conflict handling"""
+            docs_remaining = list(all_doc_ids)
+            while docs_remaining:
+                doc_id = random.choice(docs_remaining)
+                try:
+                    sg_doc = await sg.get_document(
+                        sg_db, doc_id, "_default", "_default"
+                    )
+                    if (
+                        sg_doc is None
+                        or sg_doc.body.get("sg_updates", 0) >= num_updates
+                    ):
+                        docs_remaining.remove(doc_id)
+                        continue
+
+                    updated_body = sg_doc.body.copy()
+                    updated_body["sg_updates"] = updated_body.get("sg_updates", 0) + 1
+                    updated_body["updates"] = updated_body.get("updates", 0) + 1
+                    await sg.update_documents(
+                        sg_db,
+                        [DocumentUpdateEntry(doc_id, sg_doc.revid, updated_body)],
+                        "_default",
+                        "_default",
+                    )
+                except CblSyncGatewayBadResponseError as e:
+                    if e.code == 409:  # Conflict
+                        continue  # Retry
+                    raise
+                await asyncio.sleep(0.01)  # Small delay to normalize rate
+
+        def update_from_sdk() -> None:
+            """Update documents from SDK side with CAS"""
+            docs_remaining = list(all_doc_ids)
+            while docs_remaining:
+                doc_id = random.choice(docs_remaining)
+                sdk_doc = cbs.get_document(bucket_name, doc_id, "_default", "_default")
+                if sdk_doc is None or sdk_doc.get("sdk_updates", 0) >= num_updates:
+                    docs_remaining.remove(doc_id)
+                    continue
+
+                # Ensure no _sync metadata in SDK docs
+                assert "_sync" not in sdk_doc, f"SDK doc {doc_id} contains _sync"
+
+                try:
+                    sdk_doc["sdk_updates"] = sdk_doc.get("sdk_updates", 0) + 1
+                    sdk_doc["updates"] = sdk_doc.get("updates", 0) + 1
+                    cbs.upsert_document(
```

Comment:
this concurrent write from SDK can result in errors. There is a possibility that a SDK write can overwrite an existing SG write which could result in inconsistent results. I think the right way of handling concurrent writes in python SDK is documented [here](https://docs.couchbase.com/python-sdk/current/howtos/concurrent-document-mutations.html)